### DISCUSSION TITLE
[SOLVED] Database disk image is malformed. Any way to recover from this?

### DISCUSSION THREAD

**Discussion by chrisretusn** [martinrotter/rssguard#661] [Q&A]
Slackware64-current system using KDE, rssguard 4.1.2

It appears my rssguard database is corrupt. I notice there are more that one database. I am not to worried about the feed contents, I am looking to not loose all of my feeds URLS. I do have a saved opm file but it a bit old. Here are the debug messages on starting rssguard from Konsole in pastebin: https://pastebin.com/KbT7mdGQ

I noted there are more that one database, one in ~/.config/RSS Guard 4/database/database.db and others in ~/.local/share/RSS Guard/QtWebEngine/

Thanks!

Edit: I have backups that go back a few months. What database do I need to replace or would just replacing the contents in each of those directories?

**Comment by chrisretusn** [comment]
I guess you must be busy or I missed the FAQ about this. Today I figured things out. I ran I made a copy of ~/.config/RSS Guard 4/database/database.db ran:

`qlite3 database.db ".recover" | sqlite3 new.db`

renamed new.db to database.db and replaced that copying with ~/.config/RSS Guard 4/database/database.db. I am now am back up with RSS Guard.

If that would have failed I would have restored from backups. I did give that a shot first, it worked, but I waited to long. My backups are daily x7, weekly x 4, monthly s3 , had to restore from weekly 1 which was a bit old. That why I opted for the recover option.

**Reply by martinrotter** [reply]
Hi, I simply forgot about it. Your recovery method is right. :)

**Reply by chrisretusn** [reply]
No problem, I noted you have been busy. 

Is there a specific place where the feed URL's are stored? It looks like ~/.config/RSS Guard 4/database/database.db stores the feed contents.

**Reply by martinrotter** [reply]
"database.db" stores all articles or feeds related data, except application-wide settings which are in "config.ini"

"database.db" contains feeds URLS in table "Feeds", column "source".
